pytorch_lightning/callbacks/pruning.py

```diff
@@ -0,0 +1,282 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""
+Pruning
+^^^^^^^
+
+Perform model pruning.
+
+"""
+from copy import deepcopy
+from typing import Callable, List, Optional, Tuple, Union
+
+from torch import nn
+from torch.nn.modules.container import ModuleDict, ModuleList
+
+from pytorch_lightning.callbacks.base import Callback
+from pytorch_lightning.core.lightning import LightningModule
+from pytorch_lightning.utilities import _PYTORCH_PRUNE_AVAILABLE, rank_zero_warn
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+if _PYTORCH_PRUNE_AVAILABLE:
+    import torch.nn.utils.prune as pytorch_prune
+
+
+_PYTORCH_PRUNING_FUNCTIONS = {
+    "ln_structured": pytorch_prune.ln_structured,
+    "l1_unstructured": pytorch_prune.l1_unstructured,
+    "random_structured": pytorch_prune.random_structured,
+    "random_unstructured": pytorch_prune.random_unstructured,
+}
+
+_PYTORCH_PRUNING_METHOD = {
+    "ln_structured": pytorch_prune.LnStructured,
+    "l1_unstructured": pytorch_prune.L1Unstructured,
+    "random_structured": pytorch_prune.RandomStructured,
+    "random_unstructured": pytorch_prune.RandomUnstructured,
+}
+
+
+def check_parameters_to_prune(
+    pl_module: LightningModule,
+    parameters_to_prune: Optional[List],
+    parameters: List[str] = ["weight"]
+) -> List:
+
+    is_parameters_to_prune_none = parameters_to_prune is None
+    current_modules = [m for m in pl_module.modules() if not isinstance(m, (LightningModule, ModuleDict, ModuleList))]
+
+    if parameters_to_prune is None:
+        parameters_to_prune = []
+        for p in parameters:
+            for m in current_modules:
+                param = getattr(m, p, None)
+                if param is not None:
+                    parameters_to_prune.append((m, p))
+
+    if isinstance(parameters_to_prune, list) and not is_parameters_to_prune_none:
+
+        if all(isinstance(p, (list, tuple)) and len(p) == 2 for p in parameters_to_prune):
+
+            missing_modules = []
+            missing_parameters = []
+
+            for module, param_name in parameters_to_prune:
+                if module not in current_modules:
+                    missing_modules.append(module)
+                    continue
+
+                parameter = getattr(module, param_name)
+
+                if parameter is None:
+                    missing_parameters.append(parameter)
+
+            if len(missing_modules) > 0 or len(missing_parameters) > 0:
+                raise MisconfigurationException(
+                    "Ths provided parameters_to_tune doesn't exist in the model. "
+                    f"Found mismatching modules: {missing_modules} and missing_parameters: {missing_parameters}"
+                )
+
+    return parameters_to_prune
+
+
+class PruningCallback(Callback):
```

Comment:
naming nit: other callbacks haven't included `*Callback` in the name. should we be consistent here and simply name the class `Pruning` ?

Reply:
Hey @ananthsub. Great catch ! Renamed `ModelPruning`.